### PR TITLE
Envelop Blogpost

### PR DESCRIPTION
**Pull request by dotansimha**

### CODE REVIEW

File: pages/blog/introducing-envelop.mdx [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+
+**Today we are super excited to share with you a new open-source library we’ve been working on for the past few months!**
+
+![Envelop](/blog-assets/envelop/definition.png)
+
+## TL;DR
+
+- Envelop aims to be **The GraphQL Plugin system** ([envelop.dev](https://www.envelop.dev/))
+- Envelop is not a GraphQL server, it's just a wrapper on top of the GraphQL engine.
+- Make “hard” **GraphQL capabilities easy** by installing **powerful plugins** (Caching, Tracing with Prometheus/DataDog/newRelic/Sentry/OpenTelemetry/ApolloTracing, Loggers, GraphQL-Jit, Persisted Queries, Security with rate-limit/depth-limit/Auth0 and many others from the [Plugins Hub](https://www.envelop.dev/plugins))
```
Comment (Urigo) [inline pages/blog/introducing-envelop.mdx:24]: Add links to specific filters for each plugin types

File: pages/blog/introducing-envelop.mdx [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+**Makes sense, right?** Because if you have control of all the execution pipeline, you can easily create very sophisticated plugins that implement things that were missing before with GraphQL, without changing/forking GraphQL.
+
+![Envelop](https://www.envelop.dev/assets/home-communication.png)
+
+## Getting Started
+
+To get started with **Envelop**, make sure you understand the other requirements that you need:
+
+- You need a GraphQL schema - it doesn’t matter how you created it (either with GraphQL core library, makeExecutableSchema, or any code-first / schema-first frameworks)
+- You need a HTTP server - it doesn’t matter what type of HTTP server.
```
Comment (Urigo) [inline pages/blog/introducing-envelop.mdx:104]: like express, Fastify, Koa AWS Lambda or others

**Review by Urigo — CHANGES_REQUESTED** [review]

**Comment by dotansimha** [comment]
@Urigo please review again

File: pages/blog/introducing-envelop.mdx [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+const getEnveloped = envelop({
+  plugins: [
+    /// ... other plugins ...,
+    myPlugin,
+  ],
+});
+```
+
+**[You can find here the complete plugins documentation](https://www.envelop.dev/docs/plugins)**
+
```
Comment (Urigo) [inline pages/blog/introducing-envelop.mdx:242]: What other ideas do you think you would come up with?

Here is an example of another cool plugin coming soon - [Operation Migration plugin](https://github.com/dotansimha/envelop/pull/94) - prevent GraphQL breaking changes by creating GraphQL Migrations!!

Comment (Urigo) [reply pages/blog/introducing-envelop.mdx:242]: or another one - [public and private schema filter](https://github.com/n1ru4l/graphql-public-schema-filter) - filter your GraphQL schema operations using directives

File: pages/blog/introducing-envelop.mdx [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+GraphQL is also widely adopted in the JAMStack world - and libraries that offer GraphQL out of the box migrate to Envelop to simplify parts of the code, and to allow their users to extend the GraphQL layer in a simple way.
+
+[Redwood](https://redwoodjs.com/) is a great example. We start with a small suggestion PR, and the Redwood team was open for new ideas - so [now you can use envelop if you are a Redwood user](https://community.redwoodjs.com/t/migrate-your-redwood-graphql-api-with-graphql-helix-and-envelop/2264) !
+
+[Here is a thread about why Redwood now gives you the option](https://community.redwoodjs.com/t/using-graphql-envelop-helix-in-redwood-v0-35/2276) to replace Apollo Server with [GraphQL-Helix](https://github.com/contrawork/graphql-helix) + `Envelop`.
+
+During that process, we also start to work with other frameworks and support them with that: [Loopback](https://github.com/PabloSzx/loopback-next/pull/1), [NestJS](https://github.com/PabloSzx/nestjs-graphql/pull/1), [Parse](https://github.com/parse-community/parse-server/pull/7394) and others.
```
Comment (Urigo) [inline pages/blog/introducing-envelop.mdx:296]: @PabloSzx will open these PRs, add descriptions there and then link it here.
Should we add logos to make it nicer?

**Review by Urigo — CHANGES_REQUESTED** [review]

File: pages/blog/introducing-envelop.mdx (hunk 4 above)
Comment (Urigo) [reply pages/blog/introducing-envelop.mdx:296]: yes I'm saying once @PabloSzx will open them publicly (I think he is back tomorrow?) then we'll link them there with logos?

File: pages/blog/introducing-envelop.mdx [hunk 5]
```diff
@@ -278,10 +278,13 @@ For example, one of the Envelop plugins ([`useExtendedValidation`](https://www.e
...
 
-Here are some additional examples:
+Here are some additional examples for cool new plugins:
 
 - `perEventContextResolver`: suggested in [this PR](https://github.com/graphql/graphql-js/pull/2485), and almost [available in envelop](https://github.com/dotansimha/envelop/pull/183).
 - `@oneOf`: suggested in [this PR](https://github.com/graphql/graphql-spec/pull/825), and now [available in envelop](https://github.com/dotansimha/envelop/blob/main/packages/plugins/extended-validation/README.md#union-inputs-oneof).
+- [Migrate operations](https://github.com/dotansimha/envelop/pull/94) - a new suggestion for migration GraphQL operation during `parse`, which allows simpler flow for introducing breaking changes.
+- [Public schema filter](https://github.com/n1ru4l/graphql-public-schema-filter) - for creating a simple GraphQL schema that can be used for public APIs based on existing GraphQL schema.
+- [`useOperationFieldPermissions`](https://www.envelop.dev/plugins/use-operation-field-permissions) - a plugin that allows you to check if the fields quertied in an operation are allowed for a user, before execution starts.
```
Comment (n1ru4l) [inline pages/blog/introducing-envelop.mdx:287]:
```suggestion
- [`useOperationFieldPermissions`](https://www.envelop.dev/plugins/use-operation-field-permissions) - a plugin that allows you to check if the fields queried in an operation are allowed for a user before execution starts.
```

File: pages/blog/introducing-envelop.mdx [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+- **Solve once and share across the ecosystem** - Each plugin works with any HTTP server or deployment (Express/Fastify/Netlify/VercelAWS Lambda/Azure Functions/Cloudflare Workers/Google Cloud Functions) and any schema builder (SDL, Apollo Federation, Nexus, TypeGraphQL and others)
```
Comment (Urigo) [inline pages/blog/introducing-envelop.mdx:25]: vercel/AWS Lambda

File: pages/blog/introducing-envelop.mdx [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+- **Framework for Frameworks** - Envelop will become the new basis for GraphQL Frameworks. [It's already availble if you are using RedwoodJS](https://community.redwoodjs.com/t/using-graphql-envelop-helix-in-redwood-v0-35/2276), and we have PRs open for Loopback, NestJS, Parse and others.
+- **"Babel for GraphQL"** - Envelop also aims to be the "enrichment layer" for GraphQL. You can use any new GraphQL Features today (@defer/@stream, @live queries, OneOf and any open RFC already today, even if graphql-js has not yet implemented or released it)
+
```
Comment (kamilkisiela) [inline pages/blog/introducing-envelop.mdx:28]: Is it really true? I think `@defer/@stream` requires `graphql@defer-stream-tag` release

Comment (n1ru4l) [reply pages/blog/introducing-envelop.mdx:28]: Defer/stream is wrong. Envelop is compatible with it though and we have an example setup with helix

Comment (Urigo) [reply pages/blog/introducing-envelop.mdx:28]: I would leave it as is.  when people will want to use it, they can see how to do it and it is possible.
For an announcement blog post I think it's ok

File: pages/blog/introducing-envelop.mdx [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+![Envelop](/blog-assets/envelop/envelop.png)
+
+> Very initial draft of what Envelop is.
+
+By creating these plugins, you can create custom behaviour in a very easy way.
+
+Let’s try to break a few plugins and understand how it works:
+
+- `useLogger` - hooks into the “before” of all phases, and just does `console.log`.
+- `useTimin`g - hooks into “before” and “after” of all phases, measures times, and then prints it.
```
Comment (Urigo) [inline pages/blog/introducing-envelop.mdx:88]: fix the `

File: pages/blog/introducing-envelop.mdx [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+    useGraphQLJit(),
+  ],
+});
+```
+
+> By using `useParserCache` we make sure to parse every unique operation only once.
+> By using `useValidationCache` we make sure to validate every unique operation only once.
+> By using `useGraphQLJit` we replace the default execute function with a [just-in-time implementation](https://github.com/zalando-incubator/graphql-jit).
+
+While working without clients, we saw that many pieces of code can be moved into an Envelop plugin, and shared with the community. That created tons of plugins that you can now use quickly, without implementing it on your own for that specific project!
```
Comment (Urigo) [inline pages/blog/introducing-envelop.mdx:197]: without = with our
?

**Pull request merged**